File: xijinhui_window/XiJinHuiWindow/Models/Category.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace XiJinHuiWindow.Models
{
    public class Category: HasNameLanguages
    {
        [JsonPropertyName("id")]
        public int? Id { get; set; }

        [JsonPropertyName("category_code")]
        public string CategoryCode { get; set; }

        [JsonPropertyName("remark")]
        public string Remark { get; set; }

        [JsonPropertyName("category_photo")]
        public string CategoryPhoto { get; set; }

        [JsonPropertyName("printer_id")]
        public int? PrinterId { get; set; }

        [JsonPropertyName("deleted_at")]
        public string DeletedAt { get; set; }

        [JsonPropertyName("created_at")]
        public string CreatedAt { get; set; }

        [JsonPropertyName("updated_at")]
        public string UpdatedAt { get; set; }

        [JsonPropertyName("printer")]
        public Printer Printer { get; set; }

        [JsonPropertyName("is_animal_part")]
        public bool IsAnimalPart { get; set; }

        [JsonPropertyName("is_drink")]
        public bool IsDrink { get; set; }
    }

}

File: xijinhui_laravel/app/Models/Category.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Category extends Model
{
    use HasFactory;

    protected $table = 'categories';
    protected $fillable = [
        'category_code',
        'english_name',
        'chinese_name',
        'khmer_name',
        'remark',
        'category_photo',
        'printer_id',
        'is_animal_part',
        'deleted_at',
        'created_at',
        'updated_at',
    ];

    protected $hidden = [
        'created_at',
        'deleted_at',
        'updated_at',
    ];
    protected $casts = [
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
        'is_animal_part' => 'boolean',
        'is_drink' => 'boolean',
    ];

    public function products()
    {
        return $this->hasMany(FoodDish::class, 'category_id', 'id');
    }

    public function parent()
    {
        return $this->belongsTo(Category::class, 'parent_id', 'id');
    }

    public function children()
    {
        return $this->hasMany(Category::class, 'parent_id', 'id');
    }

    public function printer()
    {
        return $this->belongsTo(Printer::class, 'printer_id', 'id');
    }
}
